include/oneapi/dpl/pstl/hetero/utils_hetero.h

```diff
@@ -148,6 +150,28 @@ struct acc_handler_count
         return (__predicate(acc[gidx]) ? 1 : 0);
     }
 };
+
+template <typename _T>
+struct __decay_with_tuple_specialization
+{
+    using type = ::std::decay_t<_T>;
+};
+
+template <typename... _Args>
+struct __decay_with_tuple_specialization<oneapi::dpl::__internal::tuple<_Args...>>
+{
+    using type = oneapi::dpl::__internal::tuple<::std::decay_t<_Args>...>;
+};
+
+template <typename... _Args>
+struct __decay_with_tuple_specialization<::std::tuple<_Args...>>
+{
+    using type = ::std::tuple<::std::decay_t<_Args>...>;
+};
+
+template <typename... _Args>
+using __decay_with_tuple_specialization_t = typename __decay_with_tuple_specialization<_Args...>::type;
+
```

Comment:
Is there a reason this is here in `utils_hetero?`  It seems like this should be accessible to any backend.  On first inspection, perhaps it belongs in `tuple_impl.h`?